sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -81,10 +81,8 @@ class QueryExecution(
   }
 
   lazy val sparkPlan: SparkPlan = tracker.measurePhase(QueryPlanningTracker.PLANNING) {
-    SparkSession.setActiveSession(sparkSession)
-    // TODO: We use next(), i.e. take the first plan returned by the planner, here for now,
-    //       but we will implement to choose the best plan.
     // Clone the logical plan here, in case the planner rules change the states of the logical plan.
+    QueryExecution.createSparkPlan(sparkSession, planner, optimizedPlan.clone())
     planner.plan(ReturnAnswer(optimizedPlan.clone())).next()
```

Comment:
Shouldn't we remove this line?

Reply:
Thanks, good catch, missed removing this line.